packages/x-charts/src/models/axis.ts

```diff
@@ -140,6 +140,11 @@ export interface ChartsYAxisProps extends ChartsAxisProps {
 }
 
 export interface ChartsXAxisProps extends ChartsAxisProps {
+  /**
+   * The minimum gap in pixels between two tick labels.
+   * @default 8
+   */
+  minTickLabelGap?: number;
```

Comment:
Added this only for the x-axis because the y-axis doesn't accept discrete data.

Reply:
what do you mean that the y-axis doesn't accept discrete data? 🤔 

Wouldn't it depend on the chart type and orientation?

Reply:
> Is that assumption wrong?

Yes, and we have a nice counter example for that: the heatmap which is made of two band axes

If you only want to add something to some scale type (point and band in your case) you have the `AxisScaleConfig`.

Notice this property will only apply on the x-axis because it's the only one to hide some labels.
The reason behind that is not the continuouse/categorical data. It's because x-axis labels are more likely to overflow since they are impacted by labels length

Reply:
Good points, hadn't thought of that.

So, in theory, the `minTickLabelGap` can also be applied to the y-axis. However, it seems unlikely we'll need it because of the reason you delineated, so I'm going to assume we're only going to implement this for the x-axis for now.